google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactoryTest.java

```diff
@@ -60,16 +76,35 @@
   private WatchdogProvider watchdogProvider;
   private ApiClock apiClock;
   private BigtableDataSettings defaultSettings;
+  private int port;
+
+  private final BlockingQueue<Attributes> attributes = new LinkedBlockingDeque<>();
 
   @Before
   public void setUp() throws IOException {
     service = new FakeBigtableService();
 
-    serviceHelper = new FakeServiceHelper(service);
-    serviceHelper.start();
+    try (ServerSocket ss = new ServerSocket(0)) {
+      port = ss.getLocalPort();
+    }
+
+    fakeServer =
+        ServerBuilder.forPort(port)
+            .addService(service)
+            .addTransportFilter(
```

Comment:
can we instead move this logic to `FakeServiceHelper`?

Reply:
👍 Moved to `FakeServiceHelper`.